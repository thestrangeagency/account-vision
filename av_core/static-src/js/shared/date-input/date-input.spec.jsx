import React from 'react';
import { shallow } from 'enzyme';

import DateInput from './date-input';

describe('DateInput', () => {
  let wrapper;
  let input;

  beforeEach(() => {
    wrapper = shallow((
      <DateInput
        id="date"
        name="date"
        placeholder="enter date here"
        required
      />
    ));

    input = wrapper.find('input');
  });

  test('it renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('it validates the date format', () => {
    const pattern = input.prop('pattern');
    const regex = new RegExp(pattern);
    expect(regex.test('10-01-2017')).toBeFalsy();
    expect(regex.test('10012017')).toBeFalsy();
    expect(regex.test('10-1-2017')).toBeFalsy();
    expect(regex.test('10/1/2017')).toBeFalsy();
    expect(regex.test('10/01/17')).toBeFalsy();
    expect(regex.test('10/01/20177')).toBeFalsy();
    expect(regex.test('110/01/2017')).toBeFalsy();
    expect(regex.test('10/01/2017')).toBeTruthy();
  });
});
