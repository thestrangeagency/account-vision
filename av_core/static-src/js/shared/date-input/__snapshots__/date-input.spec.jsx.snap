// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput it renders 1`] = `
<input
  className="form-control date-input"
  defaultValue={null}
  id="date"
  name="date"
  pattern="^^(0[1-9]|1[012])[\\\\/](0[1-9]|[12][0-9]|3[01])[\\\\/](19|20)\\\\d\\\\d$"
  placeholder="enter date here"
  required={true}
  type="text"
/>
`;
